spec/service_worker/index.bs

```diff
@@ -3109,10 +3109,11 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
       <li>Switching on <var>job</var>'s <a>worker type</a>, run these substeps with the following options:
         <dl>
         <dt><em>"<code>classic</code>"</em></dt>
-        <dd><p><a>Fetch a classic worker script</a> given <var>job</var>’s <a>serialized</a> <a href="#dfn-job-script-url">script url</a>, <var>job</var>’s <a href="#dfn-job-client">client</a>, and "<code>serviceworker</code>".</p></dd>
+        <dd><p><a>Fetch a classic worker script</a> given <var>job</var>’s <a>serialized</a> <a href="#dfn-job-script-url">script url</a>, <var>job</var>’s <a for=job>referrer</a>, null, and "<code>serviceworker</code>".</p></dd>
         <dt><em>"<code>module</code>"</em></dt>
-        <dd><p><a>Fetch a module script tree</a> given <var>job</var>’s <a>serialized</a> <a href="#dfn-job-script-url">script url</a>, "<code>omit</code>", "<code>serviceworker</code>", and <var>job</var>’s <a href="#dfn-job-client">client</a>.</p></dd>
+        <dd><p><a>Fetch a module script tree</a> given <var>job</var>’s <a>serialized</a> <a href="#dfn-job-script-url">script url</a>, "<code>omit</code>", "<code>serviceworker</code>", and some settings object.</p></dd>
```

Comment:
You can't start a worker until its global scope/settings object have been created; it doesn't really make sense. Why do you think it's necessary in this case?

It sounds like there's a bug in the referrer handling though; filing an issue at whatwg/html would be much appreciated!